api/commands/build.ts

```diff
@@ -104,16 +117,44 @@ export class Builder {
     });
   }
 
-  public buildAssertion(assertion: dataform.IAssertion) {
+  private buildAssertion(
+    assertion: dataform.IAssertion,
+    transitiveInputsByTarget: Map<string, dataform.ITarget[]>
+  ) {
     return dataform.ExecutionAction.create({
       name: assertion.name,
-      dependencyTargets: assertion.dependencyTargets,
+      transitiveInputs: this.getAllTransitiveInputs(assertion, transitiveInputsByTarget),
       dependencies: assertion.dependencies,
       type: "assertion",
       target: assertion.target,
-      tasks: this.adapter.assertTasks(assertion, this.compiledGraph.projectConfig).build(),
+      tasks: this.adapter.assertTasks(assertion, this.prunedGraph.projectConfig).build(),
       fileName: assertion.fileName,
       actionDescriptor: assertion.actionDescriptor
     });
   }
+
+  private getAllTransitiveInputs(
+    action: dataform.ITable | dataform.IOperation | dataform.IAssertion,
+    transitiveInputsByTarget: Map<string, dataform.ITarget[]>
+  ) {
+    const actionTargetKey = JSON.stringify(action.target);
+    if (!transitiveInputsByTarget.has(actionTargetKey)) {
+      let transitiveInputTargets = action.dependencyTargets;
+      for (const transitiveInputTarget of action.dependencyTargets || []) {
+        const transitiveInputAction = this.allActions.get(JSON.stringify(transitiveInputTarget));
+        if (
```

Comment:
I think this could do with a comment. I think this is terminating the search if you hit a materialized object, but shouldn't we be checking that a dataform.Table is also not of type "view" ?

Reply:
good catch!